sdk/src/ed25519_instruction.rs

```diff
@@ -33,6 +33,14 @@ pub fn new_ed25519_instruction(keypair: &ed25519_dalek::Keypair, message: &[u8])
     let signature = keypair.sign(message).to_bytes();
     let pubkey = keypair.public.to_bytes();
 
+    pairless_ed25519_instruction(pubkey, signature, message)
+}
+
+pub fn pairless_ed25519_instruction(
+    pubkey: [u8; 32],
```

Comment:
nit: let's make use of the dalek types for safety
```suggestion
    pubkey: &ed25519_dalek::PublicKey,
```